src/librustc_mir/transform/inline.rs

```diff
@@ -94,17 +94,14 @@ impl Inliner<'tcx> {
                     continue;
                 }
 
-                let self_node_id = self.tcx.hir().as_local_node_id(self.source.def_id()).unwrap();
```

Comment:
It seems that this assignment can be moved into the `if` block below; I'll update in a moment.